### PR TITLE
Restrict usage of type 'String' for return type & param types in UDFs

### PR DESCRIPTION
**Pull request by VamsiModem**

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs [hunk 1]
```diff
@@ -96,8 +99,14 @@ private IEnumerable<UserDefinedFunction> CreateUserDefinedFunctions(IEnumerable<
                 {
                     continue;
                 }
-                
-                var func = new UserDefinedFunction(udfName.Value, udf.ReturnType.GetFormulaType()._type, udf.Body, udf.IsImperative, udf.Args);
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs:100]: >GetFormulaType [](http://example.com/codeflow?start=81&length=14)

Why not fix GetFormulaType to return valid type here?
Why not update String FormulaType to Text?

**Review by lesaltzm — APPROVED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs (hunk 1 above)
Comment (VamsiModem) [reply src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs:100]: Modified GetFormulaType() to return valid type

**Review by lesaltzm — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**